From f53ef1f41fcc961f0590e9b31a85e255f6ff80ef Mon Sep 17 00:00:00 2001
From: Fang Yafen <yafen@iscas.ac.cn>
Date: Sun, 8 Aug 2021 21:35:07 +0800
Subject: [PATCH 2/2] patch for openEuler

---
 include/linux/iommu.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index f0e2c1f5d143..7b433c36111b 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -1206,11 +1206,6 @@ iommu_sva_bind_group(struct iommu_group *group, struct mm_struct *mm,
 	return NULL;
 }
 
-int iommu_bind_guest_msi(struct iommu_domain *domain,
-			 dma_addr_t giova, phys_addr_t gpa, size_t size)
-{
-	return -ENODEV;
-}
 static inline
 void iommu_unbind_guest_msi(struct iommu_domain *domain, dma_addr_t giova) {}
 
-- 
2.27.0

