From 2265f8fd13fc3b425b5249d3461b4d687bb1229d Mon Sep 17 00:00:00 2001
From: liyunfei <liyunfei33@huawei.com>
Date: Sun, 25 Jun 2023 09:20:20 +0800
Subject: [PATCH] turbostat clang compile support

---
 tools/power/x86/turbostat/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/power/x86/turbostat/Makefile b/tools/power/x86/turbostat/Makefile
index 92e139b9c..6a2523ad7 100644
--- a/tools/power/x86/turbostat/Makefile
+++ b/tools/power/x86/turbostat/Makefile
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
-CC		= $(CROSS_COMPILE)gcc
+CC		?= $(CROSS_COMPILE)gcc
 BUILD_OUTPUT	:= $(CURDIR)
 PREFIX		?= /usr
 DESTDIR		?=
-- 
2.28.0.windows.1

